### PR TITLE
block schema

### PR DESCRIPTION
**Pull request by iAmLaZa**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: blockchain/block_description.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+from hashlib import sha256
+
+
+def updatehash(*args):
```
[the remark this answers is not in the document]
Comment (iAmLaZa) [reply blockchain/block_description.py:4]: def updatehash(*args) -> str:

[the remark this answers is not in the document]
Comment (iAmLaZa) [reply blockchain/block_description.py:4]: def updatehash(*args) -> str:

File: blockchain/block_description.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        - data : data to be stored in the block
+        - number : the number of the block
+        - previous_hash : the hash of the previous block
+        - nonce : the nonce of the block ( number to make the hash of the block valid)
+    Methods:
+        - hash : returns the hash of the block
+        - __str__ : returns a string representation of the block
+    """
+
+    def __init__(self, number=0, previous_hash="0" * 64, data=None, nonce=0):
```
[the remark this answers is not in the document]
Comment (iAmLaZa) [reply blockchain/block_description.py:31]: def __init__(self, number: int, previous_hash:str ="0" * 64, data:object =None, nonce:int =0) -> None:

File: blockchain/block_description.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        self.data = data
+        self.number = number
+        self.previous_hash = previous_hash
+        self.nonce = nonce
+
+    def hashdata(self):
```
[the remark this answers is not in the document]
Comment (iAmLaZa) [reply blockchain/block_description.py:37]: def hashdata(self) -> str:

File: blockchain/block_description.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    ) -> None:
+        self.data = data
+        self.number = number
+        self.previous_hash = previous_hash
+        self.nonce = nonce
+
+    def hashdata(self) -> str:
+        return updatehash(self.number, self.previous_hash, self.data, self.nonce)
+
+    def __str__(self) -> str:
```
[the remark this answers is not in the document]
Comment (iAmLaZa) [reply blockchain/block_description.py:46]: """
        Returns the hash of the block's data.

        >>> block1 = Block(1, "0"*64, "data1", 0)
        >>> block1.hashdata()
        '47cf022b180526de8f7d0ec5951f00181d2a2853c68305c3c1fbdede2eaec30d'
"""

File: blockchain/block_description.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        previous_hash: str = "0" * 64,
+        data: object = None,
+        nonce: int = 0,
+    ) -> None:
+        self.data = data
+        self.number = number
+        self.previous_hash = previous_hash
+        self.nonce = nonce
+
+    def hashdata(self) -> str:
```
[the remark this answers is not in the document]
Comment (iAmLaZa) [reply blockchain/block_description.py:43]: """
        Returns the hash of the block's data.

        >>> block1 = Block(1, "0"*64, "data1", 0)
        >>> block1.hashdata()
        '47cf022b180526de8f7d0ec5951f00181d2a2853c68305c3c1fbdede2eaec30d'
"""

**Comment by cclauss** [comment]
Please read CONTRIBUTING.m for our definition of an algorithm.

**Pull request closed without merging**